File: arrays/longest_common_prefix.cpp
#include <bits/stdc++.h>
using namespace std;

string longest_common_prefix(vector<string>& strs) {
    string result = "";
    for (int i = 0; i < strs[0].size(); i++) {
        for (string word : strs) {
            if (i == word.size() || word[i] != strs[0][i]) {
                return result;
            }
        }
        result += strs[0][i];
    }
    return result;
}

int main() {
    vector<string> strs = {"flow", "flower", "flowing"};
    string result = longest_common_prefix(strs);

    cout << "Longest Common Prefix: " << result << endl;
    return 0;
}

File: arrays/longest_common_prefix.py
def longestCommonPrefix(strs):
    res = ""
    for i in range(len(strs[0])):
        for s in strs: # this is basically saying "for each character in the ith position of the elements in the array... are they the same or not?"
            if i == len(s) or s[i] != strs[0][i]:
                return res
        res += strs[0][i]

    return res

def main():
    strs = ["flower", "flow", "flight"]
    lcp = longestCommonPrefix(strs)
    print(lcp)

main()